### PR TITLE
Add property tests for CLI merge invariants (#317)

### LINKED ISSUE #317: Property-based tests for CLI merge precedence and validation invariants using proptest
## Summary

The CLI configuration merge pipeline introduced in https://github.com/leynos/netsuke/pull/273 establishes a number of invariants (layer precedence ordering, list-appending semantics, cross-field validation rules) that are currently exercised only through example-based `rstest` cases. Property-based testing would provide much stronger confidence that these invariants hold across arbitrary inputs.

## Context

PR https://github.com/leynos/netsuke/pull/273 introduced `src/cli/merge.rs` and `src/cli/config.rs` with the following invariants:

- **Merge precedence:** defaults → file layers → environment → CLI overrides
- **List-appending:** `cmds.build.targets` is appended in discovery order
- **Validation rules:** theme/no_emoji compatibility, spinner_mode/progress compatibility, output_format restriction

`proptest` is already available in the workspace. The current test suite in `tests/cli_tests/merge.rs` uses hand-picked examples only.

## Proposed work

- Generate arbitrary `CliConfig` layer values using `proptest` strategies
- Verify that merge ordering is consistently respected (e.g., a CLI value always wins over a file value for the same field)
- Verify that validation rules fire correctly across the full domain of enum combinations
- Verify list-appending semantics under arbitrary target lists

## Backlinks

- PR: https://github.com/leynos/netsuke/pull/273
- Requested by @leynos

### PR DESCRIPTION
**Pull request by leynos**
## Summary

Closes #317

Adds `proptest` coverage for the merge invariants introduced by PR #273, at the `CliConfig::merge_from_layers` boundary (no filesystem or environment mutation, so the cases run fast and in parallel):

- **Merge precedence:** for any combination of present/absent layers, the highest-precedence layer supplying `jobs` wins (CLI > env > file > defaults).
- **List appending:** append-merged lists concatenate in layer order (file → env → CLI) for arbitrary generated lists; file-supplied `cmds.build.targets` survive the merge intact.
- **Validation rules:** across the full `theme` × `no_emoji` and `spinner_mode` × `progress` domains, validation fires for exactly the documented conflicting pairs; `jobs` is accepted exactly within `1..=64`.

## Validation

- `make check-fmt` / `make lint` / `make test` — pass (37 suites; 5 new property tests green)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add property-based tests to verify CLI configuration merge invariants at the CliConfig::merge_from_layers boundary.

Tests:
- Add proptest-based coverage for layer precedence of the jobs setting across file, environment, and CLI inputs.
- Add generative tests ensuring list-merged CLI options append in file → environment → CLI order while preserving file-specified build targets.
- Add property tests validating documented conflict rules for theme/no_emoji and spinner_mode/progress combinations.
- Add property tests enforcing the accepted jobs range matches documented bounds.

### CODE REVIEW

File: tests/cli_tests/merge_property.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+proptest! {
+    /// `jobs` is accepted exactly within the inclusive range `1..=MAX_JOBS`.
+    ///
+    /// The domain sweeps from zero to twice the limit, so both rejection edges
+    /// and the whole accepted span are exercised rather than the two documented
+    /// examples alone.
+    #[test]
+    fn jobs_bounds_validation_matches_documented_rules(jobs in 0_usize..=(MAX_JOBS * 2)) {
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/cli_tests/merge_property.rs]: Addressed in `93b1e6f7`.

You are right that the range sampler could not promise any particular value was reached, and the doc comment claiming both rejection edges were "exercised" was wrong for the same reason. Both are fixed:

- Added `jobs_boundaries_match_the_documented_rule`, an `rstest` table pinning exactly the four decisive values — `0`, `1`, `MAX_JOBS`, `MAX_JOBS + 1` — each routed through `CliConfig::merge_from_layers` via the same `merge_file_layer` helper. Cases are named (`case_1_zero_is_rejected` … `case_4_over_max_is_rejected`) so a failure identifies the edge that moved.
- The expected outcome is written beside each value rather than recomputed from `jobs == 0 || jobs > MAX_JOBS`, so changing the production predicate fails these cases instead of silently redefining them.
- Kept the property sweep and corrected its comment. It is complement, not coverage: it samples the interior, which four fixed cases cannot do. This matches the existing shape in `src/runner/process/job_count.rs`, where deterministic boundary arrays sit beside a property sweep.
- No production behaviour changed; `src/` is byte-identical to the previous head.

Verified each case is individually decisive by mutation testing rather than by the tests merely passing: shifting the upper edge to \`jobs >= MAX_JOBS\` makes \`case_3_max_is_accepted\` fail alone, and moving the lower edge to \`jobs == 1\` fails \`case_1_zero_is_rejected\` and \`case_2_one_is_accepted\` by name. Both mutations were reverted; \`git diff src/\` is empty.

Full gates re-run clean on the new head: \`make check-fmt\`, \`make typecheck\`, \`make lint\`, \`make test\` (3194 passed, 5 skipped, 0 failed), with all four boundary cases present as distinct nextest entries.
